### PR TITLE
Feature 3.1.0 nodetimesync part2

### PR DESCRIPTION
**Pull request by chuwen95**

### CODE REVIEW

**Review by JimmyShi22 — APPROVED** [review]

File: bcos-sync/bcos-sync/BlockSync.cpp [hunk 1]
```diff
@@ -405,7 +405,16 @@ void BlockSync::onPeerStatus(NodeIDPtr _nodeID, BlockSyncMsgInterface::Ptr _sync
     auto statusMsg = m_config->msgFactory()->createBlockSyncStatusMsg(_syncMsg);
     m_syncStatus->updatePeerStatus(_nodeID, statusMsg);
     
-    m_config->nodeTimeMaintenance()->tryToUpdatePeerTimeInfo(_nodeID, statusMsg->time());
+    if(_syncMsg->version() < static_cast<int32_t>(BlockSyncPacketVersion::v1))
+    {
+        BLKSYNC_LOG(TRACE) << LOG_DESC("The version number of BlockSyncMsg is lower than the version number of the current node")
```
Comment (cyjseagull) [inline bcos-sync/bcos-sync/BlockSync.cpp:410]: Remove this log.

File: bcos-sync/bcos-sync/protocol/PB/BlockSyncMsgImpl.h [hunk 2]
```diff
@@ -30,7 +31,10 @@ class BlockSyncMsgImpl : virtual public BlockSyncMsgInterface
 {
 public:
     using Ptr = std::shared_ptr<BlockSyncMsgImpl>;
-    BlockSyncMsgImpl() : m_syncMessage(std::make_shared<BlockSyncMessage>()) {}
+    BlockSyncMsgImpl() : m_syncMessage(std::make_shared<BlockSyncMessage>()) 
+    {
```
Comment (cyjseagull) [inline bcos-sync/bcos-sync/protocol/PB/BlockSyncMsgImpl.h:35]: Remove this, and set the version to `BlockSyncStatusMsg` with `createBlockSyncStatusMsg`.

File: bcos-tool/bcos-tool/NodeTimeMaintenance.cpp [hunk 3]
```diff
@@ -42,14 +42,14 @@ void NodeTimeMaintenance::tryToUpdatePeerTimeInfo(
     {
         TIMESYNC_LOG(WARNING)
             << LOG_DESC("Invalid remote peer time:  too much difference from local time")
-            << LOG_KV("peer", nodeID->shortHex()) << LOG_KV("peerTime", time)
+            << LOG_KV("peer", nodeID->hex()) << LOG_KV("peerTime", time)
```
Comment (cyjseagull) [inline bcos-tool/bcos-tool/NodeTimeMaintenance.cpp:45]: Use `shortHex` is enough.

File: bcos-sync/bcos-sync/interfaces/BlockSyncMsgFactory.h [hunk 4]
```diff
@@ -41,7 +42,7 @@ class BlockSyncMsgFactory
     virtual BlockSyncStatusInterface::Ptr createBlockSyncStatusMsg(
         bcos::protocol::BlockNumber _number, bcos::crypto::HashType const& _hash,
         bcos::crypto::HashType const& _gensisHash, int64_t const time = utcTime(),
-        int32_t _version = 0)
+        int32_t _version = static_cast<int32_t>(BlockSyncMsgVersion::v1))
```
Comment (cyjseagull) [inline bcos-sync/bcos-sync/interfaces/BlockSyncMsgFactory.h:45]: Not set the default version to `v1` here, pass the version when calling `createBlockSyncStatusMsg` instead.

File: bcos-sync/bcos-sync/protocol/PB/BlockSyncMsgFactoryImpl.h [hunk 5]
```diff
@@ -40,7 +40,10 @@ class BlockSyncMsgFactoryImpl : public BlockSyncMsgFactory
 
     BlockSyncStatusInterface::Ptr createBlockSyncStatusMsg() override
     {
-        return std::make_shared<BlockSyncStatusImpl>();
+        BlockSyncStatusInterface::Ptr statusMsg = std::make_shared<BlockSyncStatusImpl>();
+        statusMsg->setVersion(static_cast<int32_t>(BlockSyncMsgVersion::v1));
```
Comment (cyjseagull) [inline bcos-sync/bcos-sync/protocol/PB/BlockSyncMsgFactoryImpl.h:44]: Remove this.

**Review by cyjseagull — APPROVED** [review]

**Pull request merged**